tests/Fixtures/SerializingArrayStore.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+/*
+ * This file is part of the %%Project%% package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Discovery\Tests\Fixtures;
+
+use Webmozart\KeyValueStore\Api\CountableStore;
+use Webmozart\KeyValueStore\Api\NoSuchKeyException;
+use Webmozart\KeyValueStore\Api\SortableStore;
+use Webmozart\KeyValueStore\Util\KeyUtil;
+
+/**
+ * Like {@link ArrayStore}, but with serialization.
+ *
+ * This class is used in the test since it behaves more like actual key-value
+ * stores backed by files or databases while still being fast (in memory).
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ */
+class SerializingArrayStore implements SortableStore, CountableStore
```

Comment:
well, it is mostly useful for testing, but this would avoid to have to reimplement it in every library wanting to test things with an in-memory storage behaving closely to persistent storages.